vlingo-reactive-messaging-patterns/src/main/java/io/vlingo/reactive/messaging/patterns/messagebus/AbstractTradingActor.java

```diff
@@ -0,0 +1,114 @@
+// Copyright © 2012-2018 Vaughn Vernon. All rights reserved.
+//
+// This Source Code Form is subject to the terms of the
+// Mozilla Public License, v. 2.0. If a copy of the MPL
+// was not distributed with this file, You can obtain
+// one at https://mozilla.org/MPL/2.0/.
+package io.vlingo.reactive.messaging.patterns.messagebus;
+
+import io.vlingo.actors.Actor;
+import io.vlingo.actors.testkit.TestUntil;
+import io.vlingo.reactive.messaging.patterns.messagebus.exception.UnexpectedMethodInvocationException;
+
+/**
+ * AbstractTradingActor
+ *
+ * @author brsg.io
+ * @since Oct 31, 2018
+ */
+public abstract class AbstractTradingActor 
+extends Actor 
+implements TradingProcessor
+{
+    private final TestUntil until;
+    private TradingProcessor tradingBus;
+    private final String applicationId = this.getClass().getSimpleName();
+    
+    public AbstractTradingActor( TestUntil until )
+    {
+        this.until = until;
+        initialize();
+    }
+    
+    public AbstractTradingActor( TestUntil until, TradingProcessor tradingBus )
+    {
+        this.until = until;
+        this.tradingBus = tradingBus;
+        initialize();
+    }
+    
+    abstract void initialize();
+
+    /**
+     * Returns the value of {@link #until}
+     *
+     * @return the value of {@link #until}
+     */
+    public TestUntil getUntil()
```

Comment:
May I suggest doing away with `get...()` and just use the name of the object you want such as `until()`? The whole `get` and `set` language has basically killed fluency in Java, and we need to take back our ground :)